src/Offer/Offer.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\CalendarSummaryV3\Offer;
+
+use DateTimeImmutable;
+
+final class Offer
+{
+    /**
+     * @var OfferType
+     */
+    private $offerType;
+
+    /**
+     * @var Status
+     */
+    private $status;
+
+    /**
+     * @var DateTimeImmutable|null
+     */
+    private $startDate;
+
+    /**
+     * @var DateTimeImmutable|null
+     */
+    private $endDate;
+
+    /**
+     * @var CalendarType|null
+     */
+    private $calendarType;
+
+    /**
+     * @var Offer[]
+     */
+    private $subEvents = [];
+
+    /**
+     * @var OpeningHour[]
+     */
+    private $openingHours = [];
+
+    public function __construct(
+        OfferType $offerType,
+        Status $status,
+        ?DateTimeImmutable $startDate = null,
+        ?DateTimeImmutable $endDate = null,
+        ?CalendarType $calendarType = null
+    ) {
+        $this->offerType = $offerType;
+        $this->status = $status;
+        $this->startDate = $startDate;
+        $this->endDate = $endDate;
+        $this->calendarType = $calendarType;
+    }
+
+    public static function fromJsonLd(string $json): self
+    {
+        $data = json_decode($json, true);
+
+        $offer = new self(
+            new OfferType(mb_strtolower($data['@type'])),
```

Comment:
Maybe add `ext-mbstring` to composer.json?

Reply:
It's probably cleaner to just do `strtolower` here